spec.html

```diff
@@ -1111,30 +1111,857 @@ <h1>Well-Known Symbols</h1>
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-number-signedRightShift">
+          <h1>Number::signedRightShift ( _x_, _y_ )</h1>
+          <emu-alg>
+            1. Let _lnum_ be ! ToInt32(_x_).
+            1. Let _rnum_ be ! ToUint32(_y_).
+            1. Let _shiftCount_ be the result of masking out all but the least significant 5 bits of _rnum_, that is, compute _rnum_ &amp; 0x1F.
+            1. Return the result of performing a sign-extending right shift of _lnum_ by _shiftCount_ bits. The most significant bit is propagated. The result is a signed 32-bit integer.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-number-unsignedRightShift">
+          <h1>Number::unsignedRightShift ( _x_, _y_ )</h1>
+          <emu-alg>
+            1. Let _lnum_ be ! ToInt32(_x_).
+            1. Let _rnum_ be ! ToUint32(_y_).
+            1. Let _shiftCount_ be the result of masking out all but the least significant 5 bits of _rnum_, that is, compute _rnum_ &amp; 0x1F.
+            1. Return the result of performing a zero-filling right shift of _lnum_ by _shiftCount_ bits. Vacated bits are filled with zero. The result is an unsigned 32-bit integer.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-number-lessThan">
+          <h1>Number::lessThan ( _x_, _y_ )</h1>
+          <emu-alg>
+            1. If _x_ is *NaN*, return *undefined*.
+            1. If _y_ is *NaN*, return *undefined*.
+            1. If _x_ and _y_ are the same Number value, return *false*.
+            1. If _x_ is *+0* and _y_ is *-0*, return *false*.
+            1. If _x_ is *-0* and _y_ is *+0*, return *false*.
+            1. If _x_ is *+&infin;*, return *false*.
+            1. If _y_ is *+&infin;*, return *true*.
+            1. If _y_ is *-&infin;*, return *false*.
+            1. If _x_ is *-&infin;*, return *true*.
+            1. If the mathematical value of _x_ is less than the mathematical value of _y_&mdash;note that these mathematical values are both finite and not both zero&mdash;return *true*. Otherwise, return *false*.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-number-equal">
+          <h1>Number::equal ( _x_, _y_ )</h1>
+          <emu-alg>
+            1. If _x_ is *NaN*, return *false*.
+            1. If _y_ is *NaN*, return *false*.
+            1. If _x_ is the same Number value as _y_, return *true*.
+            1. If _x_ is *+0* and _y_ is *-0*, return *true*.
+            1. If _x_ is *-0* and _y_ is *+0*, return *true*.
+            1. Return *false*.
+          </emu-alg>
+        </emu-clause>
+
+        <emu-clause id="sec-numeric-types-number-sameValue">
+          <h1>Number::sameValue ( _x_, _y_ )</h1>
+          <emu-alg>
+            1. If _x_ is *NaN* and _y_ is *NaN*, return *true*.
+            1. If _x_ is *+0* and _y_ is *-0*, return *false*.
```

Comment:
are these needed? if the zero lines are removed from this algorithm, it should fall through to false, no?

Reply:
I don't think it would be correct, since we have this statement into the spec: `Note that there is both a positive zero and a negative zero. For brevity, these values are also referred to for expository purposes by the symbols +0 and -0, respectively. (Note that these two different zero Number values are produced by the program expressions +0 (or simply 0) and -0.)`.
IMHO, it would very confusing if we rely on `If _x_ is the same Number value as _y_, return *true*.` to handle zero cases, since we don't consider them as the same Number value previously.

Reply:
i'm fine to keep them; just wanted to confirm that they aren't strictly necessary